lib/Basic/LangOptions.cpp

```diff
@@ -191,6 +203,24 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     llvm_unreachable("undefined architecture endianness");
   }
 
+  // Set the "_environment" platform condition.
+  switch (Target.getEnvironment()) {
+  default: break;
+  case llvm::Triple::MSVC:
+    addPlatformConditionValue("_environment", "msvc");
```

Comment:
Please add tests for these values appropriate files in `test/Parse/ConditionalCompilation`.